website/project/model.py

```diff
@@ -2343,10 +2347,12 @@ def add_contributors(self, contributors, auth=None, log=True, save=False):
         :param save: Save after adding contributor
         """
         for contrib in contributors:
-            self.add_contributor(
+            added = self.add_contributor(
                 contributor=contrib['user'], permissions=contrib['permissions'],
                 visible=contrib['visible'], auth=auth, log=False, save=False,
             )
+            if added:
+                project_signals.contributor_added.send(self, contributor=contrib['user'])
```

Comment:
Hmm, OK.

My gut feeling here is that calling `add_contributor` not triggering the associated event is unexpected behavior, and will likely result in the event not getting triggered when it should at some point (if it doesn't already, undetected).

I'm pulling down the code now and will take a look.